packages/flutter/test/cupertino/tableview_test.dart

```diff
@@ -0,0 +1,267 @@
+        // Build up to 20 cells.
+        if (index < 20) {
+          return const SizedBox(key: Key('cell'), width: double.infinity, height: 100.0);
+        } else {
+          return null;
+        }
+      };
+
+      // We need to track which cells were built so that we can confirm the
+      // lazy build behavior of CupertinoTableView.
+      final _TrackingIndexedWidgetBuilder trackingWidgetBuilder = _TrackingIndexedWidgetBuilder(
+        builder: widgetBuilder,
+      );
+
+      await tester.pumpWidget(
+        CupertinoApp(
+          home: CupertinoTableView.plainBuilder(
+            scrollController: scrollController,
+            plainChildCount: 20,
+            plainChildrenBuilder: trackingWidgetBuilder.buildWithTracking,
+          ),
+        ),
+      );
+
+      // Ensure that the 6 visible cells really are rendered.
+      expect(find.byKey(const Key('cell')).evaluate().length, 6);
+
+      // Ensure that only 9 cells have been built (the 6 on screen + 3 more
+      // due to cache extent).
+      expect(trackingWidgetBuilder.didBuildAllWidgetsInRange(0, 9), true);
+      expect(trackingWidgetBuilder.didNotBuildAnyWidgetsInRange(9, 20), true);
+
+      // Ensure that the 6 on-screen cells have the correct width and y-values
+      final Offset cell1TopRight = tester.getTopRight(find.byKey(const Key('cell')).at(0));
+      expect(cell1TopRight, const Offset(800.0, 0.0));
+
+      final Offset cell2TopRight = tester.getTopRight(find.byKey(const Key('cell')).at(1));
+      expect(cell2TopRight, const Offset(800.0, 100.0));
+
+      final Offset cell3TopRight = tester.getTopRight(find.byKey(const Key('cell')).at(2));
+      expect(cell3TopRight, const Offset(800.0, 200.0));
+
+      final Offset cell4TopRight = tester.getTopRight(find.byKey(const Key('cell')).at(3));
+      expect(cell4TopRight, const Offset(800.0, 300.0));
+
+      final Offset cell5TopRight = tester.getTopRight(find.byKey(const Key('cell')).at(4));
+      expect(cell5TopRight, const Offset(800.0, 400.0));
+
+      final Offset cell6TopRight = tester.getTopRight(find.byKey(const Key('cell')).at(5));
+      expect(cell6TopRight, const Offset(800.0, 500.0));
+
+      // Scroll all the way to the bottom and ensure that all 20 cells were built.
+      scrollController.jumpTo(1400.0);
+      await tester.pumpAndSettle();
+      expect(trackingWidgetBuilder.didBuildAllWidgetsInRange(0, 20), true);
+    });
+
+    testWidgets('Empty CupertinoTableView draws dividers throughout the viewport', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        const CupertinoApp(
+          home: CupertinoTableView.plain(
+            plainChildren: <Widget>[],
+          ),
+        ),
+      );
+
+      // Find the dividers RenderObject and retrieve a description of its state.
+      final RenderObject dividersSearchResult = tester.renderObject(find.byType(CupertinoTableViewExtraDividers));
```

Comment:
ya, I would just test against the mock canvas here. Would simplify your production classes a lot.

Reply:
I'm not familiar with the approach you're referring to.  Can you describe it?